Design/SANS/NewISISDesign/Design_Document.md

```diff
@@ -0,0 +1,350 @@
+# Design Document for ISIS SANS Backend Rewrite
+
+This document defines the design  for a new ISIS SANS back-end as required by
+MantidProject issue [#15875](https://github.com/mantidproject/mantid/issues/15875).
+
+The intention of the issue is to replace the organically grown and now difficult to
+maintain back-end of the ISIS SANS reduction work-flow.
+
+### Definition of terms
+
+| Term  | Definition|
+|------|------------|
+|ADS   | Analysis Data Service|
+| PI   | Python Interface |
+
+
+## Motivation
+
+The current SANS reduction system makes use of the so-called `ReductionSingleton`,
+which essentially stores the configurational state either directly, in `ReductionStep`s
+or in an `ISISInstrument` object. The `ReductionStep`s are deeply coupled to the
+`ReductionSingleton`, which makes it hard to unit test them
+(and probably explains why such tests do not exist).
+In addition there is a strong coupling of this `ReductionSingleton` to the
+GUI logic, which essentially renders the ISIS SANS reduction a monolithic block.
+
+A more modern approach is to use Mantid's `WorkflowAlgorithm`s. A sequence of
+algorithms perform the reduction, where each algorithm gets the information
+required for the reduction at the time it executes. The `WorkflowAlgorithm` is
+used as a unit of clearly defined work and **not** as a container for
+state information. This mechanism avoids the deep coupling as each algorithm is
+agnostic about its calling environment and the information flow is into the algorithms
+(except for resulting output workspaces).
+This creates an ideal environment for unit testing and improved documentation.
+
+
```

Comment:
**Summary of Benefits**
* Reduced cost of maintenance
* Reduced time for future feature additions
* Higher testability, and therefore greater robustness
* Better separation leading to more granular and simpler documentation